### DISCUSSION TITLE
Unable to create subscription for different monitored items/nodes

### DISCUSSION THREAD

**Discussion by AshuSingh07** [eclipse-milo/milo#1086] [General]
When subscribing to different Nodes, i am getting error while passing onItemCreated parameter to subscription.createmonitoredItems(). It works well when passing this parameter for single node subscription by using 
UaSubscription.ItemCreationCallback onItemCreated =
                (item, id) -> item.setValueConsumer(this::onSubscriptionValue);

but fails when passing the below code for multiple node subscription. Getting error as "Requiredtype: ItemCreationCallback, provided Biconsumar" . When casting OnItemCreated to (UaSubscription.ItemCreationCallback)onItemCreated , throwing error as can not cast. 

        BiConsumer<UaMonitoredItem, Integer> onItemCreated =
                (item, id) -> item.setEventConsumer(this::onEventReceived);

I am passing onItemCreated like this:

        List<UaMonitoredItem> items = subscription.createMonitoredItems(
                TimestampsToReturn.Both,
                newArrayList(powerRequest,pressureRequest,speedRequest),
                (UaSubscription.ItemCreationCallback) onItemCreated
        ).get();

**Comment by kevinherron** [comment]
Change this:
```java
    BiConsumer<UaMonitoredItem, Integer> onItemCreated =
            (item, id) -> item.setEventConsumer(this::onEventReceived);
```
to this:
```java
    UaSubscription.ItemCreationCallback onItemCreated =
            (item, id) -> item.setEventConsumer(this::onEventReceived);
```

**Reply by AshuSingh07** [reply]
Thank you . I tried but i was getting subscription value for only the last node passed in newArrayList. Then i realized that i have to pass different parameter and clienthandle value as:
        // important: client handle must be unique per item
        UInteger clientHandle = uint(clientHandles.getAndIncrement());

        MonitoringParameters powerParameters = new MonitoringParameters(
                clientHandle,
                1000.0,     // sampling interval
                null,       // filter, null means use default
                uint(10),   // queue size
                true        // discard oldest
        );

 UInteger clientHandle1 = uint(clientHandles.getAndIncrement());

        MonitoringParameters pressureParameters = new MonitoringParameters(
                clientHandle1,
                1000.0,     // sampling interval
                null,       // filter, null means use default
                uint(10),   // queue size
                true        // discard oldest
        );

UInteger clientHandle2 = uint(clientHandles.getAndIncrement());

        MonitoringParameters windSpeedParameters = new MonitoringParameters(
                clientHandle2,
                1000.0,     // sampling interval
                null,       // filter, null means use default
                uint(10),   // queue size
                true        // discard oldest
        );

**Comment by kleenxcoder** [comment]
I just ran into the same problem. @kevinherron can you provide a solution? Thx!

**Reply by kevinherron** [reply]
Please start a new discussion with relevant code and errors. This was solved if you read all responses so I don’t know what your issue is.
